content/en/blog/_posts/2025-11-XX-handling-undecryptable-resources-alpha/index.md

```diff
@@ -0,0 +1,228 @@
+layout: blog
+title: "Kubernetes v1.35: Handling Undecryptable Resources (Alpha)"
+date: 2025-11-XXT09:00:00-08:00
+draft: true
+slug: kubernetes-v1-35-handling-undecryptable-resources-alpha
+author: >
+  [Ben Petersen](https://github.com/benjaminapetersen) (Microsoft)
+---
+
+## Handling undecryptable resources: when one broken Secret ruins everything
+
+If you’ve ever rotated an encryption key a little too aggressively, miswired a
+KMS plugin, or inherited a cluster with a mysterious etcd history, you may have
+seen the most frustrating flavour of "encryption at rest": the part where *one
+malformed object* makes it impossible to list **any** objects of that type. A
+single Secret or ConfigMap that can’t be decrypted turns simple operations like
+`kubectl get secrets -n production` into a full-cluster faceplant. And because
+the API server can’t read the object, you can’t delete it either—unless you’re
+willing to go spelunking directly in etcd with all the associated "please don’t
+brick the cluster" anxiety.
+
+KEP-3926, *Handling undecryptable resources*, gives administrators a safer,
+API-native way to (a) find out which resources are failing to decrypt and (b)
+remove them when you really, truly mean it.
+
+## The problem: one bad object, many sad APIs
+
+Encryption at rest in Kubernetes works by transforming objects as they’re
+written to and read from storage. If that transformation fails on the way *out*
+of etcd—because a key is gone, the ciphertext is corrupted, or some other
+storage-layer issue—then:
+
+- Any request that needs to read that object fails.
+- A `LIST` or `WATCH` that touches the same prefix also fails, even if all
+	other objects are perfectly fine.
+- Because the API server can’t decode the object, you can’t just `kubectl
+	delete` your way out of trouble.
+
+Today, the escape hatch is: connect directly to etcd and surgically remove the
+broken record. That’s brittle, vendor‑unfriendly, and not something most SREs
+are excited to practice in production.
+
+The cluster needs a way to say, "this thing is unreadable, please show me which
+one it is" and, if the admin decides it’s safe, "delete it anyway"—all via the
+Kubernetes API.
+
+## The solution: better errors and a (very sharp) delete option
+
+The enhancement does two main things:
+
+1. **Improves errors for undecryptable resources**
+
+	 When the API server can’t read objects from storage, it now uses a specific
+	 status reason: `StorageReadError` (`StatusReasonStoreReadError`). That
+	 error:
+
+	 - Clearly signals that the failure happened while reading from the backend
+		 store.
+	 - Includes details about the resource kind and the key prefix.
+	 - Can list up to 100 specific keys that failed to read, so you can see which
+		 objects are malformed without poking at etcd directly.
+
+	 This alone makes it much easier to diagnose *what* is broken when a list or
+	 get explodes.
+
+2. **Introduces a new delete option for "I know this is dangerous" deletes**
```

Comment:
Saying it's an almost unconditional delete might make more sense to the reader. IIRC some conditions still apply, or might only apply in some cases, e.g. when the resource can be decoded.